### PR TITLE
ivy.index_add

### LINKED ISSUE #26801: ivy.index_add
**Why should this be implemented?**
- 3+ of the native frameworks have this function
- it's an NN-related function and should be as efficient as possible
- it's needed for a complex/long frontend function implementation

**Links to native framework implementations**
- [Jax](<FUNC_URL_DOCS>)
- [PyTorch](<FUNC_URL_DOCS>)
- [TensorFlow](<FUNC_URL_DOCS>)
- [NumPy](<FUNC_URL_DOCS>)

### PR DESCRIPTION
**Pull request by imsoumya18**
# PR Description

## Related Issue

Closes #26801 

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### Socials
[Twitter](https://twitter.com/imsoumya184)

### CODE REVIEW

**Comment by imsoumya18** [comment]
@saeedashrraf please review this PR

**Comment by imsoumya18** [comment]
@saeedashrraf can u please review the PR?

File: ivy/functional/ivy/experimental/manipulation.py [hunk 1]
```diff
@@ -2900,3 +2900,54 @@ def trim_zeros(
     ),
     "to_skip": ("inputs_to_ivy_arrays",),
 }
+
+
+@handle_nestable
+@handle_exceptions
+@handle_array_like_without_promotion
+@inputs_to_ivy_arrays
```
[the remark this answers is not in the document]
Comment (imsoumya18) [reply ivy/functional/ivy/experimental/manipulation.py]: Hi @vedpatwardhan thanks for the review. I have made the changes. Please have a look

[the remark this answers is not in the document]
Comment (vedpatwardhan) [reply ivy/functional/ivy/experimental/manipulation.py]: sounds good, thanks!

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Thanks for making the changes, the tests seem to be failing with [tensorflow](https://github.com/unifyai/ivy/actions/runs/7082403750/job/19273606133?pr=26934#step:4:396) which fails the tests with all backends (because `tensorflow` is the ground truth). I'll also request @NripeshN's review who'd be able to suggest the further steps , thanks @imsoumya18 😄

**Review by vedpatwardhan — COMMENTED** [review]
Hey @imsoumya18, @NripeshN would be reviewing the PR soon (as he's assigned to the PR). Thanks 😄

**Comment by imsoumya18** [comment]
> Hey @imsoumya18, @NripeshN would be reviewing the PR soon (as he's assigned to the PR). Thanks 😄

Yes, please. I have been waiting since October for this PR. If needed we can schedule a meeting. I am really worried about this one. Thank you 😄

**Comment by vedpatwardhan** [comment]
Hey @imsoumya18, sorry just realized one thing, isn't this function be the same as `ivy.scatter_nd` with `sum` being the `reduction`? Was it present on a ToDo list cuz ideally this function shouldn't have been on a ToDo list. Thanks 😄

**Comment by imsoumya18** [comment]
> Hey @imsoumya18, sorry just realized one thing, isn't this function be the same as `ivy.scatter_nd` with `sum` being the `reduction`? Was it present on a ToDo list cuz ideally this function shouldn't have been on a ToDo list. Thanks 😄

Actually I was implementing `index_add` #26394 & `index_add_` #26761 in Paddle frontend as listed as ToDo in #15037

Then, I got [this review](https://github.com/unifyai/ivy/pull/26761#pullrequestreview-1663702572) and I paused the above 2 and this issue is still pending since then. Whereas the actual 2 was completed but not merged until it is being done.

**Comment by vedpatwardhan** [comment]
> > Hey @imsoumya18, sorry just realized one thing, isn't this function be the same as `ivy.scatter_nd` with `sum` being the `reduction`? Was it present on a ToDo list cuz ideally this function shouldn't have been on a ToDo list. Thanks 😄
> 
> Actually I was implementing `index_add` #26394 & `index_add_` #26761 in Paddle frontend as listed as ToDo in #15037
> 
> Then, I got [this review](https://github.com/unifyai/ivy/pull/26761#pullrequestreview-1663702572) and I paused the above 2 and this issue is still pending since then. Whereas the actual 2 was completed but not merged until it is being done.

Got it, thanks for clarifying! I completely agree with the point @hello-fri-end made, the frontend implementations shouldn't be as long and we should do most of the heavy-lifting in the ivy api itself. However, I suspect using `ivy.scatter_nd` with the `sum` reduction might actually end up making the frontend implementation compact enough without the need of adding a new function to `ivy` itself. Could you please try using `ivy.scatter_nd` in the frontend implementation and see if the tests pass? Thanks @imsoumya18 😄

**Comment by imsoumya18** [comment]
NO MORE NEEDED

**Pull request closed without merging**